include/swift/SIL/SILModule.h

```diff
@@ -246,7 +247,7 @@ class SILModule {
   // Intentionally marked private so that we need to use 'constructSIL()'
   // to construct a SILModule.
   SILModule(ModuleDecl *M, SILOptions &Options, const DeclContext *associatedDC,
-            bool wholeModule);
+            bool wholeModule, PrimarySpecificPaths PSPs);
```

Comment:
This is definitely not the way we want to go. A SILModule is intended to be independent of inputs, just like the AST representation of a source file is only very loosely connected to the buffer and underlying file-on-disk. It's fine for SIL*Gen* to have knowledge of how the module is getting created, but this seems like a violation of purpose.

Reply:
Would you like me to investigate an alternative in which the paths are not part of the SILModule, but rather passed along as parameters where they are needed instead?

Reply:
Have moved the paths out of SILModule and into parameters.